### PR TITLE
Fix integrity issues

### PR DESCRIPTION
**Pull request by yakkomajuri**
Ignores flags that aren't active, and accompanies PostHog/posthog#4764 to fix the issue with the project being able to change.

### CODE REVIEW

File: featureflags.go [hunk 1]
```diff
@@ -107,11 +108,17 @@ func (poller *FeatureFlagsPoller) fetchNewFeatureFlags() {
...
-	poller.featureFlags = featureFlagsResponse.Results
+	newFlags := []FeatureFlag{}
+	for _, flag := range featureFlagsResponse.Results {
+		if flag.Active {
+			newFlags = append(newFlags, flag)
+		}
+	}
+	poller.mutex.Lock()
+	poller.featureFlags = newFlags
 	poller.mutex.Unlock()
```
Comment (fuziontech) [inline featureflags.go:122]: 👍

**Review by fuziontech — APPROVED** [review]

File: featureflags.go (hunk 1 above)
Comment (yakkomajuri) [reply featureflags.go:122]: issue wasn't the mutex btw - was just that we weren't filtering for active flags (in any lib 😬 )

Comment (fuziontech) [reply featureflags.go:122]: no no, I know mutex wasn't issue. This is just more clean locking on just where the mutation was happening.

>(in any lib 😬 )
🥶

Comment (yakkomajuri) [reply featureflags.go:122]: yeah indeed

**Pull request merged**